### PR TITLE
CDAP-4257 Add supported versions of Spark

### PR DESCRIPTION
**Pull request by awholegunch**
Added Spark dependency, re-organized the table alphabetically, and added additional notes.

Updated for Build 2:

- Passes [Quick Build](http://builds.cask.co/browse/CDAP-DOB97-2)
- Revised [Hadoop Compatibility page](http://builds.cask.co/artifact/CDAP-DOB97/shared/build-2/Docs-HTML/3.3.1-SNAPSHOT/en/admin-manual/hadoop-compatibility.html)

Fix for https://issues.cask.co/browse/CDAP-4257

### CODE REVIEW

File: cdap-docs/admin-manual/source/hadoop-compatibility.rst [hunk 1]
```diff
@@ -9,27 +9,31 @@ Hadoop Compatibility
 ====================
 
 Before installing the CDAP components, you must first install (or have access to) a Hadoop
-cluster with *HDFS*, *YARN*, *HBase*, and *ZooKeeper*. In order to use the ad-hoc querying
-capabilities of CDAP (Explore), you will also need *Hive*. All CDAP components can be installed on
-the same boxes as your Hadoop cluster, or on separate boxes that can connect to the Hadoop
-services.
+cluster with *HBase*, *HDFS*, *YARN*, and *ZooKeeper*. In order to use the ad-hoc querying
```
Comment (sreevatsanraman) [inline cdap-docs/admin-manual/source/hadoop-compatibility.rst]: Hive and Spark are optional components; hive is required to enable ad-hoc querying capabilities and spark is required if the CDAP application uses spark program.

File: cdap-docs/admin-manual/source/hadoop-compatibility.rst [hunk 2]
```diff
@@ -40,53 +44,68 @@ Internet access on the CDAP service nodes (or until the issue `CDAP-3957
...
++----------------+---------------------------------+---------------------------------------+
+
+For a Distributed CDAP cluster, version |release|, you *can* (optionally) install these
+Hadoop components, as required:
+
++----------------+---------------------------------+---------------------------------------+
+| Component      | Source                          | Supported Versions                    |
++================+=================================+=======================================+
+| **Hive**       | Apache                          | Version 0.12.0 through 1.1.0          |
++ (required for  +---------------------------------+---------------------------------------+
```
Comment (sreevatsanraman) [inline cdap-docs/admin-manual/source/hadoop-compatibility.rst]: Remove the text in paranthesis

**Comment by sreevatsanraman** [comment]
minor comments, fix and merge.

File: cdap-docs/admin-manual/source/hadoop-compatibility.rst (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/admin-manual/source/hadoop-compatibility.rst]: Fixed.

**Pull request merged**